app/agents/chain_of_thoughts.py

```diff
@@ -3,44 +3,134 @@
 from agents.flow_based import BaseFlowAgent
 from flow.flow import Flow, PromptNode, ToolNode, ChoiceNode, StartNode
 from prompts.flow_guidance_cot import FlowChainOfThoughts, PROMPT_START_STRING
+from guidance_tooling.guidance_programs.tools import ingest_file
+from guidance_tooling.guidance_programs.tools import clean_text
+from guidance_tooling.guidance_programs.tools import load_tools
+
+import os
+from langchain.llms import LlamaCpp
+from dotenv import load_dotenv
+import os
+from chromadb.config import Settings
+from colorama import Fore, Style
+from langchain.chains import RetrievalQA
+from langchain.llms import LlamaCpp
+
+llm = None
+valid_answers = ['Action', 'Final Answer']
+valid_tools = ["Check Question", "Google Search"]
+TEST_FILE = os.getenv("TEST_FILE")
+
+def get_llm():
+    global llm
+    if llm is None:
+        print("Loading guidance model...")
+        model_type = "LlamaCpp"
+        model_path ="/home/karajan/Downloads/open-llama-3b-q8_0.bin"
+        model_n_ctx =1000
+        target_source_chunks = 4
+        n_gpu_layers = 500
+        use_mlock = 0
+        n_batch = os.environ.get('N_BATCH') if os.environ.get('N_BATCH') != None else 512
+        callbacks = []
+        qa_prompt = ""
+        llm = LlamaCpp(model_path=model_path, n_ctx=model_n_ctx, callbacks=callbacks, verbose=False,n_gpu_layers=n_gpu_layers, use_mlock=use_mlock,top_p=0.9, n_batch=n_batch)
+    return llm
+
 
 
 class ChainOfThoughtsAgent(BaseFlowAgent):
```

Comment:
1. Please don't delete my CoT flow agent :cry:  - I  plan to re-use it in the future :)
2. You can use a new class for your implementation.
3. Since you're not using any of the Flow "nodes", you don't need to inherit from `BaseFlowAgent`. You could inherit from `BaseAgent`.